### ISSUE TITLE
[Bug]: Inability to do time-based join, and potentially to name time column different than 'ts'

### ISSUE THREAD

**Issue by samuell** [featureform/featureform#423]
### Expected Behavior

Showing some numerical values from features and labels in the console.

(Documented in detail in this repo, which contains a full bug reproduction setup: https://github.com/samuell/bugs/tree/main/20220831-ff-join-bug#readme)

### Actual Behavior

Got this stack trace:

```bash
$ python train.py 
Traceback (most recent call last):
  File "/home/sal/proj/sav/2022/bug-reproductions/20220831-ff-join-bug/train.py", line 5, in <module>
    train_data = client.training_set("traindata", "default")
  File "/home/sal/.cache/pypoetry/virtualenvs/20220831-ff-join-bug-JAwT4QYU-py3.9/lib/python3.9/site-packages/featureform/serving.py", line 49, in training_set
    return self._local_training_set(name, version)
  File "/home/sal/.cache/pypoetry/virtualenvs/20220831-ff-join-bug-JAwT4QYU-py3.9/lib/python3.9/site-packages/featureform/serving.py", line 112, in _local_training_set
    trainingset_df = pd.merge_asof(trainingset_df, df.sort_values(['ts']), direction='backward',
  File "/home/sal/.cache/pypoetry/virtualenvs/20220831-ff-join-bug-JAwT4QYU-py3.9/lib/python3.9/site-packages/pandas/util/_decorators.py", line 311, in wrapper
    return func(*args, **kwargs)
  File "/home/sal/.cache/pypoetry/virtualenvs/20220831-ff-join-bug-JAwT4QYU-py3.9/lib/python3.9/site-packages/pandas/core/frame.py", line 6259, in sort_values
    k = self._get_label_or_level_values(by, axis=axis)
  File "/home/sal/.cache/pypoetry/virtualenvs/20220831-ff-join-bug-JAwT4QYU-py3.9/lib/python3.9/site-packages/pandas/core/generic.py", line 1779, in _get_label_or_level_values
    raise KeyError(key)
KeyError: 'ts'
```

After trying to rename all the occurances of "ts" to "time" in `data.csv` and `defs.py`, and running
```
rm -rf .featureform && sleep 1 && featureform apply --local defs.py
python train.py
```
.. I instead got this:  

```bash
$ python train.py 
Traceback (most recent call last):
  File "/home/sal/proj/sav/2022/bug-reproductions/20220831-ff-join-bug/train.py", line 5, in <module>
    train_data = client.training_set("traindata", "default")
  File "/home/sal/.cache/pypoetry/virtualenvs/20220831-ff-join-bug-JAwT4QYU-py3.9/lib/python3.9/site-packages/featureform/serving.py", line 49, in training_set
    return self._local_training_set(name, version)
  File "/home/sal/.cache/pypoetry/virtualenvs/20220831-ff-join-bug-JAwT4QYU-py3.9/lib/python3.9/site-packages/featureform/serving.py", line 112, in _local_training_set
    trainingset_df = pd.merge_asof(trainingset_df, df.sort_values(['ts']), direction='backward',
  File "/home/sal/.cache/pypoetry/virtualenvs/20220831-ff-join-bug-JAwT4QYU-py3.9/lib/python3.9/site-packages/pandas/core/reshape/merge.py", line 580, in merge_asof
    op = _AsOfMerge(
  File "/home/sal/.cache/pypoetry/virtualenvs/20220831-ff-join-bug-JAwT4QYU-py3.9/lib/python3.9/site-packages/pandas/core/reshape/merge.py", line 1740, in __init__
    _OrderedMerge.__init__(
  File "/home/sal/.cache/pypoetry/virtualenvs/20220831-ff-join-bug-JAwT4QYU-py3.9/lib/python3.9/site-packages/pandas/core/reshape/merge.py", line 1623, in __init__
    _MergeOperation.__init__(
  File "/home/sal/.cache/pypoetry/virtualenvs/20220831-ff-join-bug-JAwT4QYU-py3.9/lib/python3.9/site-packages/pandas/core/reshape/merge.py", line 681, in __init__
    self._validate_specification()
  File "/home/sal/.cache/pypoetry/virtualenvs/20220831-ff-join-bug-JAwT4QYU-py3.9/lib/python3.9/site-packages/pandas/core/reshape/merge.py", line 1809, in _validate_specification
    raise MergeError(
pandas.errors.MergeError: Incompatible merge dtype, dtype('O') and dtype('O'), both sides must have numeric dtype
```

(Documented in detail in this repo, which contains a full bug reproduction setup: https://github.com/samuell/bugs/tree/main/20220831-ff-join-bug#readme)

### Steps To Reproduce

1. Make sure to have [poetry](https://python-poetry.org/) installed (or install featureform 1.2.0 in another way).
2. Run the following commands:
    ```
    git clone https://github.com/samuell/bugs.git
    cd 20220831-ff-join-bug
    poetry install
    poetry shell
    featureform apply --local defs.py
    python train.py
    ```

(Documented in detail in this repo, which contains a full bug reproduction setup: https://github.com/samuell/bugs/tree/main/20220831-ff-join-bug#readme)

### What mode are you running Featureform in?

Local

### What version of Python are you running?

3.9

### Featureform Python Package Version

1.2.0

**Comment by samuell** [comment]
Note that there seems to be two issues here:

1. Potentially a hard coded assumption for the time data column to be named `ts`?
2. Inability to merge even when time column is named `ts` because of Pandas requiring numerical values for merge.

**Comment by sdreyer** [comment]
Can you confirm the featureform package version is 1.2.0? That version on Pypi was mistakenly labeled and yanked earlier this month

**Comment by samuell** [comment]
@sdreyer Pip gives 1.2.0:
```
$ pip list | grep featureform
featureform       1.2.0
```
... but I assume that might be the wrong pypi label then. How can I check the actual version?

**Comment by samuell** [comment]
Confirmed now that I have the same behavior in Featureform 1.1.10 (`KeyError: 'ts'` and then the Pandas merge error due to `dtype('O')` when renaming the timestamp column to 'ts')

**Comment by sdreyer** [comment]
Got it. I'll take a look. 

As for the versioning, 1.2.0 is ignored with `pip install`. I'm wondering if poetry ignores the yanked status on Pypi. Regardless, we have the actual 1.2.0 coming out at the end of the week.

**Comment by sdreyer** [comment]
I believe I found the issue. We'll be pushing an update later today.

**Comment by sdreyer** [comment]
Hey @samuell , we've fixed this issue in the latest updated 1.1.12. You may need to clear and recreate the .featureform since there was a slight schema change

**Comment by samuell** [comment]
> Hey @samuell , we've fixed this issue in the latest updated 1.1.12. You may need to clear and recreate the .featureform since there was a slight schema change

Hi @sdreyer , that's awesome! I did try out the fix yesterday already, and it works like a charm. Seriously impressed by the quick lead times here :D
